API/DTOs/BeatmapSetCompactDTO.cs

```diff
@@ -0,0 +1,57 @@
+using System.ComponentModel.DataAnnotations;
+using Database.Enums;
+
+namespace API.DTOs;
+
+/// <summary>
+/// Represents a compact version of a beatmapset
+/// </summary>
+public class BeatmapSetCompactDTO
+{
+    /// <summary>
+    /// Beatmapset id
+    /// </summary>
+    public int Id { get; set; }
+
+    /// <summary>
+    /// osu! beatmapset id
+    /// </summary>
+    public long OsuId { get; set; }
+
+    /// <summary>
+    /// Id of the Player who created the set
+    /// </summary>
+    public int CreatorId { get; set; }
+
+    /// <summary>
+    /// Artist
+    /// </summary>
+    [MaxLength(512)]
+    public string Artist { get; set; } = string.Empty;
+
+    /// <summary>
+    /// Title
+    /// </summary>
+    [MaxLength(512)]
+    public string Title { get; set; } = string.Empty;
+
+    /// <summary>
+    /// Ranked status
+    /// </summary>
+    public BeatmapRankedStatus RankedStatus { get; set; }
+
+    /// <summary>
+    /// Date of ranking, if applicable
+    /// </summary>
+    public DateTime? RankedDate { get; set; }
+
+    /// <summary>
+    /// Date of submission
+    /// </summary>
+    public DateTime? SubmittedDate { get; set; }
+
+    /// <summary>
+    /// The set creator
+    /// </summary>
+    public PlayerCompactDTO Creator { get; set; } = new();
```

Comment:
```suggestion
    public PlayerCompactDTO? Creator { get; set; }
```
Bad! This is how we get gems like:
```json
"user": {
    "avatar_url": null,
    "country_code": null,
    "default_group": "default",
    "id": null,
    "is_active": false,
    "is_bot": false,
    "is_deleted": true,
    "is_online": false,
    "is_supporter": false,
    "last_visit": null,
    "pm_friends_only": false,
    "profile_colour": null,
    "username": "[deleted user]"
}
```